src/server.c

```diff
@@ -4076,12 +4076,13 @@ int processCommand(client *c) {
         if (server.current_client == NULL) return C_ERR;
 
         int reject_cmd_on_oom = is_denyoom_command;
-        /* If client is in MULTI/EXEC context, queuing may consume an unlimited
-         * amount of memory, so we want to stop that.
+        /* If client is in MULTI/EXEC context, even if no DENYOOM command is used, 
+         * we want to limit the amount of memory used for queuing.
          * However, we never want to reject DISCARD, or even EXEC (unless it
          * contains denied commands, in which case is_denyoom_command is already
          * set. */
         if (c->flags & CLIENT_MULTI &&
+            c->mstate.argv_len_sums > MULTI_MAX_BYTES_ON_OOM && 
```

Comment:
```suggestion
            c->mstate.argv_len_sums > server. proto_max_bulk_len && 
```
I'm still going to advocate this should be a config, since I think it's possible 32k will be too low.

Reply:
ok, fair enough, clients can easily consume that amount of memory without multi-exec, so i'm fine using that limit (1gb).
but then i'd remind you that for unauthenticated clients we have a limit of 16k
https://github.com/redis/redis/blob/628c0dea1b12ae28ce44a8cfa638dc784fe1bb2b/src/networking.c#L2359

Reply:
I suppose this loophole still exists today as long as the server is not at max memory